jwst/rscd/rscd_step.py

```diff
@@ -32,6 +31,18 @@ def process(self, input):
                     result.meta.cal_step.rscd = 'SKIPPED'
                     return result
 
+
+                # Check that data has the minimum number of groups/int
+                sci_ngroups = input_model.data.shape[1]     # number of groups
+                min_number = 4
+                if sci_ngroups < min_number :
+                    self.log.warning('Input file does not contain enough groups '
+                                     ' for RSCD correctionto be applied ')
+                    self.log.warning('RSCD step will be skipped')
+                    result = input_model.copy()
```

Comment:
In cases where a step is being skipped and hence the data aren't being modified in any way other than perhaps setting the cal_step meta attribute, I suggest NOT making a copy and just returning the input in place. I.e. just set input_model.meta.cal_step.rscd and return input_model. No point in making a copy of an unchanged data model.